### PR TITLE
Break the cyclic dependency between BaseTarget and UntypedTarget.

### PR DESCRIPTION
**Pull request by bjaspan**
This PR factors `createContinuousRangeTarget` out of BaseTarget into a new function, `createContinuousRangeOrUntypedTarget`, which is used by a variety of target types.  As a result, BaseTarget.ts no longer needs to import UntypedTarget (indirectly through createContinuousRange.ts).

### CODE REVIEW

**Comment by auscompgeek** [comment]
Hmm, there's a fair amount of copy paste here.

Is there a reason we keep `createContinuousRange*`, `BaseTarget`, and `UntypedTarget` in separate files?

**Comment by pokey** [comment]
Yeah I'm thinking maybe the right answer is to just put those in the same file? That seems less onerous than this approach. Still don't love it, but at least better than this.

@auscompgeek what's your opinion on circular deps?

**Comment by bjaspan** [comment]
I tried merging BaseTarget and UntypedTarget as well.  It turns out that UntypedTarget depends on PlainTarget (via TokenInsertionRemovalBehavior) which also depends on BaseTarget.  So, to solve the problem that way, we'd have to merge BaseTarget.ts, UntypedTarget.ts, and PlainTarget.ts into a single file.  I'm fine with that solution; I went with this one only because I guessed that you would all prefer it to merging three files.

**Comment by bjaspan** [comment]
Hold on. I don't think this PR even really fixes the problem. There is still a cyclic dependency between UntypedTarget.ts and createContinuousRange.ts.

I think we have to merge the three files.

**Comment by bjaspan** [comment]
I removed the cyclic dependency between UntypedTarget.ts and createContinuousRange.ts.  It would be nice if we had a tool to create a dependency graph, but I believe what we have now is:

* BaseTarget.ts depends on nothing relevant.
* createContinuousRange.ts depends on nothing relevant.
* UntypedTarget.ts depends on BaseTarget.ts and createContinuousRange.ts.
* All target types depend on BaseTarget.ts, many depend on UntypedTarget.ts, and some depend directly on createContinuousRange.ts. 
*

**Comment by bjaspan** [comment]
https://github.com/cursorless-dev/cursorless/pull/2047 is the merging-files approach.

**Comment by bjaspan** [comment]
I've eliminated the copy-paste by introducing a new abstract base template class, CommonTarget, which provides the createContinuousRangeOrUntypedTarget functionality. By exporting it from UntypedTarget, the import graph described in my previous comment is preserved.

File: .vscode/settings.json [hunk 1]
```diff
@@ -25,5 +25,9 @@
   "files.eol": "\n",
   "typescript.enablePromptUseWorkspaceTsdk": true,
   "typescript.tsdk": "node_modules/typescript/lib",
-  "eslint.workingDirectories": [{ "pattern": "packages/*/" }]
+  "eslint.workingDirectories": [
+    {
+      "pattern": "packages/*/"
+    }
+  ]
```
Comment (pokey) [inline .vscode/settings.json:32]: strange this formatting would have changed cc/ @auscompgeek

File: packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
 } from "../targetUtil/insertionRemovalBehaviors/TokenInsertionRemovalBehavior";
+import { createContinuousRange } from "../targetUtil/createContinuousRange";
+import { isSameType } from "../../util/typeUtils";
+
+/**
+ * An abstract target base class who createContinueRangeTarget method returns either
+ * a createContinuousRange if this and endTarget have the same type or
+ * otherwise returns an UntypedTarget from this to endTarget.
+ */
+export abstract class CommonTarget<
```
Comment (pokey) [inline packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts:17]: I think I may call this `CommonBaseTarget`? Idk wdyt @auscompgeek

**Review by pokey — COMMENTED** [review]
Yeah this looks find modulo a couple comments. Would be great to get high-level feedback from @auscompgeek if possible, but if too busy we can move forward anyway

File: .vscode/settings.json (hunk 1 above)
Comment (auscompgeek) [reply .vscode/settings.json:32]: :shrug: looks like prettier is fine with either too? weird that this is the _only_ change to this file

File: packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts (hunk 2 above)
Comment (auscompgeek) [reply packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts:17]: `CommonBase` sounds like a bit of a tautology to me 😄

File: .vscode/settings.json (hunk 1 above)
Comment (bjaspan) [reply .vscode/settings.json]: I removed this change.

File: packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts (hunk 2 above)
Comment (bjaspan) [reply packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts:17]: What about `CommonContinuousRangeTarget`?  Any name is fine with me. :-)

**Comment by pokey** [comment]
closing in favour of https://github.com/cursorless-dev/cursorless/pull/2074

**Pull request closed without merging**